public/app/plugins/panel/geomap/editor/MapCenterEditor.tsx

```diff
@@ -23,63 +19,56 @@ export const MapCenterEditor: FC<StandardEditorProps<MapCenterConfig, any, Geoma
     return centerPointRegistry.selectOptions(ids);
   }, [value?.id]);
 
+  useEffect(() => {
+    if (!context.builder) {
+      return;
+    }
+
+    const category = ['Map View'];
+    // let optionsToRemove: string[] = [];
+
+    if (views.current[0].value === MapCenterID.Coordinates) {
+      context.builder
+        .addSliderInput({
+          path: 'view.center.lat',
+          name: 'Latitude',
+          category,
+          settings: {
+            min: -90,
+            max: 90,
+          },
+        })
+        .addSliderInput({
+          path: 'view.center.lon',
+          name: 'longitude',
+          category,
+          settings: {
+            min: -180,
+            max: 180,
+          },
+        });
+      // optionsToRemove = ['view.center.lat', 'view.center.lon'];
+    }
+
+    // return () => {
+    //   if (context.builder && optionsToRemove.length > 0) {
+    //     for (let i = 0; i < optionsToRemove.length; i++) {
+    //       console.log('remove', optionsToRemove[i]);
+    //       context.builder.remove(optionsToRemove[i]);
+    //     }
+    //   }
+    // };
```

Comment:
Played with unregistering options when they are no longer relevant, but I think it doesn't make sense. I only do not trigger PanelOptionsChanged event when option that already exists in the registry tries to be registered again - See OptionsUIBuilder